daffodil-test/src/test/resources/org/apache/daffodil/layers/midBitsToEnd.tdml

```diff
@@ -0,0 +1,80 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+<tdml:testSuite xmlns:tdml="http://www.ibm.com/xmlns/dfdl/testData" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns:dfdl="http://www.ogf.org/dfdl/dfdl-1.0/" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:fn="http://www.w3.org/2005/xpath-functions"
+  xmlns:ex="http://example.com" xmlns:tns="http://example.com" defaultRoundTrip="true">
+
+  <tdml:defineSchema name="s1" elementFormDefault="unqualified">
+    <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>
+    <dfdl:defineFormat name="general">
+      <dfdl:format ref="ex:GeneralFormat" />
+    </dfdl:defineFormat>
+    <dfdl:defineFormat name="bits8_9_toEnd">
+      <dfdl:format ref="ex:general" layerTransform="midBitsToEnd" layerTransformArgs="7 2" layerLengthKind="explicit" layerLength="3" layerLengthUnits="bytes" />
+    </dfdl:defineFormat>
+    <dfdl:format ref="ex:general" />
+
+    <xs:element name="root">
+      <xs:complexType>
+        <xs:sequence dfdl:ref="tns:bits8_9_toEnd">
+          <xs:element name="transformedContent" type="xs:hexBinary" dfdl:lengthKind="explicit" dfdl:length="3" />
+        </xs:sequence>
+      </xs:complexType>
+    </xs:element>
+  </tdml:defineSchema>
+
+  <tdml:parserTestCase name="midBitsToEnd1" root="root" model="s1">
+    <tdml:document>
+      <tdml:documentPart type="bits"><![CDATA[
+      0000 0011
+      1100 0000
```

Comment:
This is taking 2 bits in Most-significant-bit-first positions 1 and 2, and moving them to the end, shifting everything toward the most-significant bit. The 2 extracted bits end up as the least-significant bits of the final byte. 

That's the MSBF-version of this feature. You need a test that does LSBF don't you?

Reply:
Added LSBF test